### PR TITLE
Features/#545 ch4 buses abroad

### LINKED ISSUE #545: Create CH4 buses im neighbouring countries (eGon2035)
Currently, the the gas pipes connected to neighbouring countries are defined as links, but these have one bus which is not defined (the one abroad). This should be fixed.

**TODOs**

- [ ] Define one CH4 bus per neighbouring country. See https://github.com/openego/eGon-data/blob/features/%23138-integrate-pysa-eur-sec/src/egon/data/datasets/electrical_neighbours.py 
- [ ] Redefine the links modelling the cross border gas pipes.
- [ ] Optional (to be done later): Redefine the capacities of the cross border gas pipes with https://entsog.eu/sites/default/files/2021-07/ENTSOG_TYNDP_2020_Annex_C2_Capacities_per_country.xlsx

### PR DESCRIPTION
**Pull request by AmeliaNadal**
Fixes #545  . 

## Before merging into `dev`-branch, please make sure that

- [x] the `CHANGELOG.rst` was updated.
- [x] new and adjusted code is formated using `black` and `isort`.
- [x] the `Dataset`-version is updated when existing datasets are adjusted. 
- [x] the branch was merged into the `continuous-integration/run-everything-over-the-weekend`- branch.
- [x] the workflow is running successful in `test mode`.
- [ ] the workflow is running successful in `Everything` mode.

### CODE REVIEW

File: src/egon/data/datasets/gas_grid.py [hunk 1]
```diff
@@ -213,16 +214,108 @@ def insert_CH4_nodes_list(gas_nodes_list):
...
+
+    Returns
+    -------
+    gdf_abroad_buses : dataframe
+        Dataframe containing the gas in the neighbouring countries and one in the center of Germany in test mode
+    """
+    if scn_name == "eGon2035":
+        main_gas_carrier = "CH4"
+    elif scn_name == "eGon100RE":
+        main_gas_carrier = "H2"
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/gas_grid.py:232]: The `main_gas_carrier` could be part of the scenario_parameters table.

File: src/egon/data/datasets/gas_grid.py [hunk 2]
```diff
@@ -260,9 +353,11 @@ def insert_gas_pipeline_list(gas_nodes_list):
     # Select the links having at least one bus in Germany
     gas_pipelines_list = gas_pipelines_list[
         gas_pipelines_list["country_code"].str.contains("DE")
-    ]  # A remplacer evtmt par un test sur le NUTS0 ?
+    ]
 
     # Remove links disconnected of the rest of the grid, until the SciGRID_gas data has been corrected.
+    # TODO: automatic test for disconnected links
+    # TODO: remove link test if length = 0
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/gas_grid.py:360]: Are these ToDos still pending?

File: src/egon/data/datasets/gas_grid.py (hunk 1 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_grid.py]: I changed this, but it still need to be integrated to the CI branch.

File: src/egon/data/datasets/gas_grid.py [hunk 3]
```diff
@@ -213,17 +215,103 @@ def insert_CH4_nodes_list(gas_nodes_list):
...
+    ----------
+    scn_name : str
+        Name of the scenario
+
+    Returns
+    -------
+    gdf_abroad_buses : dataframe
+        Dataframe containing the gas in the neighbouring countries and one in the center of Germany in test mode
+    """
+    main_gas_carrier = scenario_parameters.gas(scn_name)["main_gas_carrier"]
```
Comment (nesnoj) [inline src/egon/data/datasets/gas_grid.py:230]: @AmeliaNadal @ClaraBuettner 
To be consistent, shouldn't the scenario parameters read by using [`egon.data.datasets.scenario_parameters.get_sector_parameters()`](https://github.com/openego/eGon-data/blob/0730acfae19def9381f184b28f3304c5f7c0be3c/src/egon/data/datasets/scenario_parameters/__init__.py#L101)?

File: src/egon/data/datasets/scenario_parameters/parameters.py [hunk 4]
```diff
@@ -82,10 +82,10 @@ def gas(scenario):
     """
 
     if scenario == "eGon2035":
-        parameters = {}
+        parameters = {"main_gas_carrier": "CH4"}
 
     elif scenario == "eGon100RE":
-        parameters = {}
+        parameters = {"main_gas_carrier": "H2"}
```
Comment (ClaraBuettner) [inline src/egon/data/datasets/scenario_parameters/parameters.py:88]: Please update the version number of ScenarioParameters

File: src/egon/data/datasets/gas_grid.py (hunk 3 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_grid.py]: Thanks for the remarks, I made the requiered change.

File: src/egon/data/datasets/gas_grid.py (hunk 2 above)
Comment (AmeliaNadal) [reply src/egon/data/datasets/gas_grid.py:359]: Yes, they are. Here is the new issue: https://github.com/openego/eGon-data/issues/590

**Review by ClaraBuettner — APPROVED** [review]

**Pull request merged**